Sources/JExtractSwiftLib/JNI/JNISwift2JavaGenerator+SwiftThunkPrinting.swift

```diff
@@ -293,11 +293,43 @@ extension JNISwift2JavaGenerator {
     let tryClause: String = decl.isThrowing ? "try " : ""
 
     // Regular parameters.
-    var arguments = [String]()
+    var arguments: [String] = [String]()
+    var indirectVariables: [(name: String, lowered: String)] = []
+    var int32OverflowChecks: [String] = []
+
     for (idx, parameter) in nativeSignature.parameters.enumerated() {
       let javaParameterName = translatedDecl.translatedFunctionSignature.parameters[idx].parameter.name
       let lowered = parameter.conversion.render(&printer, javaParameterName)
       arguments.append(lowered)
+
+      parameter.indirectConversion.flatMap {
+        indirectVariables.append((javaParameterName, $0.render(&printer, javaParameterName)))
+      }
+
+      switch parameter.conversionCheck {
+        case .check32BitIntOverflow:
+          int32OverflowChecks.append(parameter.conversionCheck!.render(&printer, "indirect_\(javaParameterName)"))
+        case nil:
+          break
+      }
+    }
+
+    // Make indirect variables
+    for (name, lowered) in indirectVariables {
+      printer.print("let indirect_\(name) = \(lowered)")
```

Comment:
we have a naming convention for variables to suffix with `$` so we don't clash with user defined variables; It's very unlikely anyone would do some "indirect_bla" but please let's be consistent.

This could be `name$indirect`?